code/datums/components/surgery_initiator.dm

```diff
@@ -0,0 +1,278 @@
+/// Allows an item to be used to initiate surgeries.
+/datum/component/surgery_initiator
+	/// If present, this surgery TYPE will be attempted when the item is used.
+	/// Useful for things like limb reattachments that don't need a scalpel.
+	var/datum/surgery/forced_surgery
+
+	/// If true, the initial step will be cancellable by just using the tool again. Should be FALSE for any tool that actually has a first surgery step.
+	var/can_cancel_before_first = FALSE
+
+	/// If true, can be used with a cautery in the off-hand to cancel a surgery.
+	var/can_cancel = TRUE
+
+/datum/component/surgery_initiator/Initialize(datum/surgery/forced_surgery = null)
```

Comment:
is the `null` declaration here necessary? If you don't specific a default value it should automatically be null if no value is given in new.

Reply:
I guess so, though I wanted to make it clear that the argument was optional. I suppose I could also just add documentation to the proc for that (which I should have done anyway :U )